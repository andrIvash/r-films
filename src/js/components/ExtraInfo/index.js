// @flow
import React from 'react';
import helpers from '../../helpers';
import type { View } from '../../flow-types.js';

type Props = {
  data: string,
  view: View
}

const ExtraInfo = (props: Props) => {
  const { view, data } = props;
  return (
    <div className='extra-info'>
      <div className='extra-info__content'>
        { view === helpers.views.POSTER ?
          `Films by ${data} genre` : `${data} movies found` }
      </div>
    </div>
  );
};

export default ExtraInfo;
