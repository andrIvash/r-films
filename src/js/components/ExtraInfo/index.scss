.extra-info__content {
  font-family: $sign-font;
  font-size: rem(11px);
  font-weight: 700;
  color: $color-background;
}