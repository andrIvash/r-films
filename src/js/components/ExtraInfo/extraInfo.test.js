import React from 'react';
import 'jest-styled-components';
import ExtraInfo from './index';

let props = {};

describe('ExtraInfo', () => {
  beforeEach(()=> {
    props = {
      data: '',
      view: '',
    };
  });
  it('should be defined', () => {
    expect(ExtraInfo).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = render(
      <ExtraInfo {...props} /> );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render right values in POSTER view', () => {
    props.data = 'Drama';
    props.view = 'POSTER';

    const wrapper = shallow(
      <ExtraInfo {...props} /> );

    expect(wrapper.find('ExtraContent').children().text())
      .toBe('Films by Drama genre');
  });
  it('should render right values in COMMON view', () => {
    props.data = '9';
    props.view = 'COMMON';

    const wrapper = shallow(
      <ExtraInfo {...props} /> );

    expect(wrapper.find('ExtraContent').children().text())
      .toBe('9 movies found');
  });
});
