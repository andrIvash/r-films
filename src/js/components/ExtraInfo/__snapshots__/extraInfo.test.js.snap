// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtraInfo should render correctly 1`] = `
<div
  class="extra-info"
>
  <div
    class="extra-info__content"
  >
     movies found
  </div>
</div>
`;
